### DISCUSSION TITLE
Sdk container builds as kaniko replacement

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/sdk-container-builds#579] [General]
We use kaniko mainly to remove docker dependency and root user requirements. It can also be configured to push each intermediate layer to a cache repo but it takes a dockerfile to perform a build. I think to fully supersede this setup, container builds should either provide some level of feature parity (primarily around caching and some niceties like push retries) or ways to integrate with an existing kaniko setup.

/cc @baronfel

**Comment by baronfel** [comment]
I have some good news for you :)

By default we make nonroot images - you have to change [ContainerUser](https://github.com/dotnet/sdk-container-builds/blob/main/docs/ContainerCustomization.md#containeruser) to enable root usage.

We do cache the intermediate layers, and when we push layers to the destination registry we check if they already exist on the remote registry before attempting to push. 

So I think we are essentially at parity with the features you mentioned today!
